exonum/src/events/noise/wrapper.rs

```diff
@@ -60,6 +63,12 @@ impl NoiseWrapper {
     }
 
     pub fn read_handshake_msg(&mut self, input: &[u8]) -> Result<(usize, Vec<u8>), NoiseError> {
+        if input.len() < NOISE_MIN_HANDSHAKE_MESSAGE_LENGTH
+            || input.len() > NOISE_MAX_MESSAGE_LENGTH
```

Comment:
...And this restriction seems to be unreachable by design if the method is invoked from `NoiseHandshake::read_handshake_msg`.